pkg/db/strategy.go

```diff
@@ -122,12 +135,25 @@ func (s *Strategy) Create(ctx context.Context, object types.Object) (types.Objec
 		return nil, err
 	}
 
+	var (
+		fieldNames []string
+		vals       []any
+	)
+	if o, ok := object.(types.Fields); ok {
+		for _, f := range o.FieldNames() {
+			fieldNames = append(fieldNames, f)
```

Comment:
I know this is premature optimization, but can you allocate a slice. of the exact size needed

Reply:
Done.